lib/collections/addon/guid/route.ts

```diff
@@ -80,7 +83,7 @@ export default class Guid extends Route {
         }
 
         return {
-            taskInstance: this.loadModel.perform(guid) as Promise<TaskInstanceResult>,
+            taskInstance: taskFor(this.loadModel).perform(guid) as Promise<TaskInstanceResult>,
```

Comment:
without taskFor `this.loadModel` is simply an async function, therefore no `this.loadModel.perform()` would error. This happens because TS doesn't yet allow decorators to change the type of the thing they are decorating.

Reply:
I was referring to the `Promise<TaskInstanceResult>`. Do we still need that?

Reply:
Ah no, we don't.